### ISSUE TITLE
Hiding a large data array; hyperlinking text without showing the link in the table

### ISSUE THREAD

**Issue by adstore** [olifolkerd/tabulator#810]
1. In your 1000-record DOM example, the source data for the table is not contained in the file code, and I can't see a reference to an external file.  How do I do that?  [In your other examples the data records are shown explicitly in the script html file.]  

2. Can text values - eg. Names, Businesses, etc. - not be directly hyperlinked?  The link formatter option requires you to put in a full path external link column which is kind of messy in the user display, and would be instantly harvested by opportunistic users.  I couldn't see any examples of this being done.  (See the scenario in www.adscanusa.com/tabulator/test1.html.

Thanks!

**Comment by olifolkerd** [comment]
Hey @adstore 

Most of the examples don't show how the data is loaded because there are a number of different ways to pull data into Tablulator and each demo source code only shows the feature it is demonstrating, Have a look at the [Set Table Data Documentation](http://tabulator.info/docs/3.3#set-data) for more info.

You can format your data in any way you like using custom formatters. if you store the url for the link in a ***link*** property in your row data object and the title in the ***link_title*** property, then you could use a custom formatter to do this:

```js
//define custom formatter
linkFormatter = function(cell, formatterParams){
    var data = cell.getData();

    return $("<a href='" + cell.getValue() + "'>" + data.link_title + "</a>");
}

//set formatter in column definition
{title:"Link", field:"link", formatter:linkFormatter},
```

**Comment by olifolkerd** [comment]
Hey @adstore 

I have just pushed some updates to the 3.4 branch to improve the link formatter, there are now additional ***formatterParams*** you can pass in to change the label for the link:

See issue #800 for more details

These will be available in the 3.4 release in a couple of weeks time.

Cheers

Oli

**Comment by olifolkerd** [comment]
Tabulator version 3.4 has now been officially release!

You can find details of the new release in the [News Section](http://tabulator.info/news)

Full documentation of all the new features can be found [Here](http://tabulator.info/docs/3.4)

Cheers

Oli :)
